### PR TITLE
TST Fix #265: de-flake test_bgl_unfair via ridge regularization

### LINKED ISSUE #265: test_bgl_unfair fails consistently on Windows with Python3.5
Example: https://dev.azure.com/responsibleai/fairlearn/_build/results?buildId=3657&view=logs&jobId=a846d25a-e32c-5640-1b53-e815fab94407&j=f2ae15cd-30d2-5fb2-07f7-558e9be6b891&s=778b61a2-25e7-5600-2cb0-dd2dc8e23627

The affected code:
```
assert logging_all_close([[3.2, 11.2],
                              [-3.47346939, 10.64897959],
                              [-2.68, 10.12],
                              [-1.91764706, 9.61176471],
                              [-1.18461538,  9.12307692],
                              [-0.47924528,  8.65283019],
                              [0.2, 0.7]],
                             all_predict)
```
and only the first entry fails.
```
E        +  where False = logging_all_close([[3.2, 11.2], [-3.47346939, 10.64897959], [-2.68, 10.12], [-1.91764706, 9.61176471], [-1.18461538, 9.12307692], [-0.47924528, 8.65283019], ...], [array([ 3.03010885, 11.2       ]), array([-3.47346939, 10.64897959]), array([-2.68, 10.12]), array([-1.91764706,  9.61176471]), array([-1.18461538,  9.12307692]), array([-0.47924528,  8.65283019]), ...])

test\unit\reductions\grid_search\test_grid_search_regression.py:72: AssertionError
---------------------------- Captured stdout call -----------------------------
a mismatches:  [3.2]
b mismatches:  [3.03010885]
mismatch indices:  (array([0], dtype=int64), array([0], dtype=int64))
```

### PR DESCRIPTION
**Pull request by romanlutz**
## Description

`test_bgl_unfair`'s first grid point has been platform-flaky since 2020 (issue #265). The root cause is mathematical, not numerical: at the extreme `BoundedGroupLoss` grid point the Lagrangian effectively zeros out one group, and the remaining weighted least-squares system is **rank-deficient** (the design has 3 columns but the kept group's `sensitive_features` is constant and `constant_ones_feature` is constant overall, so effective rank is 2). An infinite family of solutions all achieve the same minimum loss, and *which one* `lstsq` returns depends on the underlying LAPACK driver. The previous workaround papered over this with an `or` clause matching either of the two observed answers.

This PR implements @MiroDudik's 2020 suggestion: switch the test to a regularized inner estimator so the optimization problem is **strictly convex** and the minimizer is uniquely defined on every numpy/BLAS/LAPACK combination.

**Approach.** A small `alpha` knob is added to the test-only `LeastSquaresRegressor` in `simple_learners.py`, implemented via an augmented `np.linalg.lstsq` system (no new dependencies). `test_bgl_unfair` is switched from `sklearn.linear_model.LinearRegression` to `LeastSquaresRegressor(alpha=1e-6)`. The `or` workaround and the `# Reflects #265` TODO are removed. Sibling tests in the file (`test_bgl_unmitigated_same`, `test_bgl_lagrange_specifications`) are untouched; they don't trigger the degeneracy.

**Why `alpha=1e-6` is safe.** Far above the FP noise floor (~2.2e-16) so the regularization wins against numerical noise, far below the scale of meaningful coefficients (~O(10)) so the interior grid points barely move. Verified: interior predictions shift by ~1e-7, comfortably inside `np.allclose`'s default tolerance, so the published golden values still match. No production code is changed.

**Correctness verification.**
- Math: the augmented `lstsq` formulation matches the closed-form `(X^T W X + alpha I)^-1 X^T W y` to machine epsilon (~1e-15).
- Mechanism: explicit construction of the affine family of OLS optima shows non-uniqueness; ridge collapses it to a single point.
- Behaviour: `test_bgl_unfair` x 50 reps on both `numpy==1.26.4` and `numpy==2.4.6` -- all green. Full reductions suite (3895 tests) passes.

Note that bumping `numpy>=2` would also mask the symptom on most current platforms, but it doesn't fix the underlying degeneracy and re-exposes the test to any future BLAS change. Tracked separately; out of scope here.

Fixes: #265

## Tests
- [ ] no new tests required
- [x] new tests added
- [x] existing tests adjusted

New `test/unit/reductions/exponentiated_gradient/test_simple_learners.py` adds 5 focused unit tests for the `alpha` knob, including a closed-form correctness check and a rank-deficient-design check that directly mirrors the #265 failure mode. `test_bgl_unfair` is adjusted to use the regularized estimator and drops the `or` workaround.

## Documentation
- [ ] no documentation changes needed
- [x] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

Added a bug-fix entry to `docs/user_guide/installation_and_version_guide/v0.15.0.rst`.

## Screenshots

N/A

## CI notes

The failing checks on the latest run (`27653444104`) are unrelated to this PR's diff and are caused by flaky dataset-download infrastructure, not by the test/docs change here.

- `all-deps (windows-latest, 3.11)` -- **failure**. The 4 failing tests are all OpenML dataset downloads in `test/unit/datasets/test_datasets.py`: `test_dataset_as_bunch[fetch_credit_card-True/False]` and `test_dataset_as_X_y[fetch_credit_card-True/False]`. Each fails with `RuntimeWarning: Invalid cache, redownloading file`, emitted by scikit-learn's OpenML fetcher while downloading the credit-card dataset (`https://api.openml.org/data/v1/download/21854402/default-of-credit-card-clients.arff`). Because `pyproject.toml` sets `filterwarnings = ["error", ...]`, that network/cache `RuntimeWarning` is promoted to a hard failure. This is non-deterministic remote-download/cache flakiness and should clear on rerun.
- `all-deps (ubuntu-latest, 3.13)` -- **cancelled**, not an independent failure. It was stopped by fail-fast off the windows-3.11 job.

This PR touches only 4 files (`simple_learners.py`, `test_simple_learners.py`, `test_grid_search_regression.py`, and the `v0.15.0.rst` changelog); it does not touch `test/unit/datasets/` or any dataset-fetching code. On the same failing windows-latest / 3.11 job, this PR's own tests all passed: the 5 `test_simple_learners.py` tests and both `test_bgl_unfair[True]`/`test_bgl_unfair[False]` are green in the log. No code change is warranted; a rerun of the dataset-download job is the appropriate remedy.

### CODE REVIEW

**Review by adrinjalali — APPROVED** [review]
Looks legit and only touches tests, so LGTM. I just don't think it needs a changelog since it doesn't really change anything user facing.

File: docs/user_guide/installation_and_version_guide/v0.15.0.rst [hunk 1]
```diff
@@ -27,6 +27,7 @@ Bug fixes
   attribute :code:`constraints_`: :pr:`1632` by
   :user:`BALOGUN DAVID TAIWO <BALOGUN-DAVID>`.
 * Fixed :code:`fetch_diabetes_hospital` so that :code:`as_frame=False` returns numpy arrays instead of raising. The data is now always fetched as a pandas DataFrame internally and converted to numpy when needed. The skipped tests and the obsolete :code:`test_fetch_diabetes_hospital_as_ndarray_raises_value_error` test were removed: :pr:`1636` by :user:`Richard Ogundele <richardogundele>`.
+* De-flaked :code:`test_bgl_unfair` by adding a small ridge regularization (via a new :code:`alpha` parameter on the test-only :code:`LeastSquaresRegressor`) so the inner least-squares optimum is uniquely defined at the extreme :code:`BoundedGroupLoss` grid points. This removes a long-standing platform-dependent failure where the unregularized problem was rank-deficient and LAPACK could pick different equally-optimal solutions, closing :issue:`265`: :pr:`1652` by :user:`Roman Lutz <romanlutz>`.
```
Comment (romanlutz) [inline docs/user_guide/installation_and_version_guide/v0.15.0.rst]:
```suggestion
```
